apis/python/src/tiledbsoma/_dataframe.py

```diff
@@ -29,6 +29,8 @@
 from .options._tiledb_create_options import TileDBCreateOptions
 
 _UNBATCHED = options.BatchSize()
+# TODO: in Python ≥3.10, this can be `Tuple[Any, Any] | List[Any]`. In Python <3.10 it raises `TypeError: unsupported operand type(s) for |: '_GenericAlias' and '_GenericAlias'`
+Domain = Sequence[Optional[Sequence[Any]]]
```

Comment:
This just looks like it’s related to the syntax of types, not the type itself: [`|` as a type union operator](https://docs.python.org/3.10/whatsnew/3.10.html#pep-604-new-type-union-operator) was introduced in 3.10. This can be represented as `Union[Tuple[Any, Any], List[Any]]`.